src/modules/form/components/DynamicForm.tsx

```diff
@@ -45,6 +45,7 @@ export interface DynamicFormProps
     FormActionProps,
     'disabled' | 'loading' | 'onSubmit' | 'onSaveDraft'
   > {
+  clientId?: string;
```

Comment:
This is okay, but maybe we can add a comment saying that it only gets used if this form is an Assessment. It's a nonsense prop if the form is being used to edit a Project record, for example.

Another option would be to accept an object like RecordPickerDialogProps, so its flexible. I usually tend toward that approach, but I think it can be confusing in other ways, so I don't feel like either way is better.

Reply:
Good point, I'll add a comment. I agree that this isn't ideal. Perhaps in the future it could be refactored to provide a more friendly interface that allows for the picker to be composed into just the forms that need it.